drivers/video/video_esp32_dvp.c

```diff
@@ -339,6 +339,37 @@ static int video_esp32_get_ctrl(const struct device *dev, unsigned int cid, void
 	return video_get_ctrl(cfg->source_dev, cid, value);
 }
 
+static int video_esp32_flush(const struct device *dev, enum video_endpoint_id ep, bool cancel)
+{
+	struct video_esp32_data *data = dev->data;
+	struct video_buffer *vbuf = NULL;
+
+	if (cancel) {
+		if (data->is_streaming) {
+			video_esp32_set_stream(dev, false);
+		}
+		if (data->active_vbuf) {
+			k_fifo_put(&data->fifo_out, data->active_vbuf);
+			data->active_vbuf = NULL;
+		}
+		while (vbuf = k_fifo_get(&data->fifo_in, K_NO_WAIT) != NULL) {
+			k_fifo_put(&data->fifo_out, vbuf);
+			#ifdef CONFIG_POLL
+			if (data->signal_out) {
+				k_poll_signal_raise(data->signal_out, VIDEO_BUF_ABORTED);
+			}
+			#endif
+			}
```

Comment:
One `}` too much?
The `#ifdef`s are typically 0-indented to help with this, but it is not part of the coding style [[1](https://docs.zephyrproject.org/latest/contribute/style/code.html), [2](https://kernel.org/doc/html/latest/process/coding-style.html)], so probably no need to bother!.

Reply:
Why not using the recommended snippet?
```c
if (IS_ENABLED(CONFIG_POLL) && data->signal) { 
 	k_poll_signal_raise(data->signal, VIDEO_BUF_ABORTED); 
 } 
 ```

Reply:
Ok, I see now. `struct k_poll_signal *signal;` not defined as you commented.